### PR TITLE
Allow server to be auto selected from config

### PR DESCRIPTION
**Pull request by csduarte**
#### Summary
This PR allows for the server to be auto selected for the client device based on the `DefaultServerUrl` property in the config.json. The following property has been added to the config.json:

AutoSelectServerUrl
Default is false.
Set to the true to allow for the `DefaultServerUrl` to be auto selected for the client.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux) (please link)
- [x] Has UI changes
- [ ] Includes text changes and localization file updates

@jarredwitt will coordinate modifications

### CODE REVIEW

**Comment by jarredwitt** [comment]
@enahum removed the easter egg. We'll figure something else out.

**Comment by enahum** [comment]
Do you want me to review it either way?

File: app/components/auto_select_url_easter_egg.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import React, {PureComponent} from 'react';
```
Comment (enahum) [inline app/components/auto_select_url_easter_egg.js]: I thought you removed this

File: app/screens/login_options/login_options.js [hunk 2]
```diff
@@ -192,33 +193,35 @@ class LoginOptions extends PureComponent {
...
-                    style={[GlobalStyles.subheader, {fontWeight: 'bold', marginTop: 10}]}
-                    id='mobile.login_options.choose_title'
-                    defaultMessage='Choose your login method'
-                />
-                {this.renderEmailOption()}
-                {this.renderLdapOption()}
-                {this.renderGitlabOption()}
-                {this.renderSamlOption()}
-            </ScrollView>
+            <View style={style.container}>
```
Comment (enahum) [inline app/screens/login_options/login_options.js]: Why this container view? just asking...

File: assets/base/config.json [hunk 3]
```diff
@@ -1,4 +1,5 @@
 {
+    "AppName": "Mattermost",
```
Comment (enahum) [inline assets/base/config.json]: Looks like this is the setting you had missing in the about dialog PR

**Comment by jarredwitt** [comment]
@enahum changes made. Actually removed the easter egg component this time 😝

File: app/screens/select_server/select_server.js [hunk 4]
```diff
@@ -205,20 +208,22 @@ class SelectServer extends PureComponent {
...
-                            onSubmitEditing={this.onClick}
-                            style={[GlobalStyles.inputBox, allowOtherServers ? {} : {backgroundColor: '#e3e3e3'}]}
-                            autoCapitalize='none'
-                            autoCorrect={false}
-                            keyboardType='url'
-                            placeholder={{id: 'mobile.components.select_server_view.siteUrlPlaceholder', defaultMessage: 'https://mattermost.example.com'}}
-                            returnKeyType='go'
-                            underlineColorAndroid='transparent'
-                        />
+                        {!Config.AutoSelectServerUrl &&
```
Comment (hmhealey) [inline app/screens/select_server/select_server.js]: How does this render if AutoSelectServerUrl is set? It is just "Enter Server URL" with a "Proceed" button below it?

Comment (jarredwitt) [reply app/screens/select_server/select_server.js]: ah yeah, the "Enter Server URL" should be hidden as well. I'll make that change.

**Comment by hmhealey** [comment]
@jarredwitt Another conflict with the build config

**Pull request merged**